src/frontends/tensorflow_common/src/op/truncate_div.cpp

```diff
@@ -0,0 +1,41 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "common_op_table.hpp"
+#include "openvino/op/ceiling.hpp"
+#include "openvino/op/divide.hpp"
+#include "openvino/op/equal.hpp"
+#include "openvino/op/floor.hpp"
+#include "openvino/op/less.hpp"
+#include "openvino/op/mod.hpp"
+#include "openvino/op/select.hpp"
+
+using namespace std;
+using namespace ov::opset10;
+
+namespace ov {
+namespace frontend {
+namespace tensorflow {
+namespace op {
+OutputVector translate_truncate_div_op(const NodeContext& node) {
+    default_op_checks(node, 2, {"TruncateDiv"});
+    auto x = node.get_input(0);
+    auto y = node.get_input(1);
+
+    auto res = make_shared<Divide>(x, y);
+    auto mod_res = make_shared<Mod>(x, y);
+    auto is_mod_zero = make_shared<Equal>(mod_res, create_same_type_const_scalar(x, 0));
+    auto is_res_negative = make_shared<Less>(res, create_same_type_const_scalar(x, 0));
+
+    auto modified_res = make_shared<Select>(is_res_negative, make_shared<Ceiling>(res), make_shared<Floor>(res));
+
+    auto final = make_shared<Select>(is_mod_zero, res, modified_res);
+
+    set_node_name(node.get_name(), final);
+    return final->outputs();
```

Comment:
```suggestion
    auto res = make_shared<Divide>(x, y);
    auto is_res_negative = make_shared<Less>(res, create_same_type_const_scalar(x, 0));
    auto final_res = make_shared<Select>(is_res_negative, make_shared<Ceiling>(res), make_shared<Floor>(res));

    set_node_name(node.get_name(), final_res);
    return final_res->outputs();
```

Reply:
@siddhant-0707, I have simplified a bit a sub-graph because Ceil(x) == Floor(x) if x is integer so no need to compute mod:)